internal/nss/group/group.go

```diff
@@ -93,12 +85,12 @@ var groupIterationCache *cache.Cache
 
 // StartEntryIteration open a new cache for iteration.
 // This needs to be called prior to calling NextEntry and be closed with EndEntryIteration.
-func StartEntryIteration(ctx context.Context) error {
+func StartEntryIteration(ctx context.Context, cacheOpts ...cache.Option) error {
```

Comment:
Since I can't comment on a line that's not in the diff, I'll add the comment right here...
The race is all about the (group|passwd|shado)IterationCache variable that we have in internal/nss/ subpackages. We have some mechanisms to avoid racing in a single execution (like this StartEntryIteration), but this does not prevent racing in parallel executions (as we've seen in the tests report). We could use a mutex to avoid this, but this is kind of problem will not happen in a real use of the lib, so we might have to do that in the tests that rely on this functionality rather than on the lib itself.